### PR TITLE
Customer stall UI

### PR DESCRIPTION
**Pull request by talvasconcelos**
Refactor ot the Market UI for Nostr

### CODE REVIEW

File: static/components/chat-dialog/chat-dialog.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      },
+      async signEvent(event) {
+        if (this.account.privkey) {
+          event.sig = await NostrTools.signEvent(event, this.account.privkey)
+        } else if (this.account.useExtension && this.hasNip07) {
+          event = await window.nostr.signEvent(event)
+        }
+        return event
+      },
+      timeFromNow(time) {
```
Comment (motorina0) [inline static/components/chat-dialog/chat-dialog.js:150]: maybe move this function to `utils.js`

File: views.py [hunk 2]
```diff
@@ -20,3 +20,21 @@ async def index(request: Request, user: User = Depends(check_user_exists)):
...
+@nostrmarket_ext.get("/market", response_class=HTMLResponse)
+async def market(
+    request: Request,
+    stall_id: str = Query(None),
+    product_id: str = Query(None),
+    merchant_pubkey: str = Query(None),
+):
+    return nostrmarket_renderer().TemplateResponse(
+        "nostrmarket/market.html",
+        {
```
Comment (motorina0) [inline views.py:34]: - why forwards `stall_id`, `product_id` and `merchant_pubkey` back to the customer
- these values have been received on the request in the first place (therefore the customer already has them)

File: static/js/market.js [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+      await this.initNostr()
+
+      this.$q.loading.hide()
+    },
+    methods: {
+      naddr() {
+        let naddr = NostrTools.nip19.naddrEncode({
+          identifier: '1234',
+          pubkey:
+            'c1415f950a1e3431de2bc5ee35144639e2f514cf158279abff9ed77d50118796',
```
Comment (motorina0) [inline static/js/market.js:147]: hard-coded value

File: static/js/market.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        inputPubkey: null,
+        inputRelay: null,
+        activePage: 'market',
+        activeStall: null,
+        activeProduct: null
+      }
+    },
+    computed: {
+      filterProducts() {
+        let products = this.products
```
Comment (motorina0) [inline static/js/market.js:61]: todo: fiter on relay side

File: static/js/market.js [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+        if (this.activeStall) {
+          products = products.filter(p => p.stall_id == this.activeStall)
+        }
+        if (!this.searchText || this.searchText.length < 2) return products
+        return products.filter(p => {
+          return (
+            p.name.includes(this.searchText) ||
```
Comment (motorina0) [inline static/js/market.js:68]: should be case insensitive
```js
const searchText = this.searchText.toLowerCase()
p.name.toLowerCase().includes(searchText)
```

File: static/components/chat-dialog/chat-dialog.js [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+            })
+            relay.on('error', () => {
+              console.debug(`failed to connect to ${relay.url}`)
+            })
+
+            await relay.connect()
+            let pub = relay.publish(event)
+            pub.on('ok', () => {
+              console.debug(`${relay.url} has accepted our event`)
+              relay.close()
```
[the remark this answers is not in the document]
Comment (talvasconcelos) [reply static/components/chat-dialog/chat-dialog.js:110]: copy paste typo. fixed

File: views.py (hunk 2 above)
Comment (talvasconcelos) [reply views.py:34]: left overs from the initial plan on having query params. fixed

File: static/js/market.js [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+            p.description.includes(this.searchText) ||
+            p.categories.includes(this.searchText)
+          )
+        })
+      },
+      stallName() {
+        return this.stalls.find(s => s.id == this.activeStall)?.name || 'Stall'
```
Comment (motorina0) [inline static/js/market.js:75]: - maybe just have `this.activeStall` be a stall object
- then use `this.activeStall.id`, `{{activeStall.name}}`, etc

File: static/components/chat-dialog/chat-dialog.js (hunk 1 above)
Comment (talvasconcelos) [reply static/components/chat-dialog/chat-dialog.js:150]: Fixed!

File: static/js/market.js (hunk 4 above)
Comment (talvasconcelos) [reply static/js/market.js:61]: maybe have a second one...

File: static/js/market.js (hunk 5 above)
Comment (talvasconcelos) [reply static/js/market.js:68]: fixed

File: static/js/market.js (hunk 7 above)
Comment (talvasconcelos) [reply static/js/market.js:75]: When i set these variables, activeStall, activeProduct, etc... there's no stalls or products yet! it just parses them from the query params. Also used to set query params when changing pages! Let's keep it for now and revisit it later, is that ok?

Comment (motorina0) [reply static/js/market.js:75]: got it, ok!

File: static/js/market.js [hunk 8]
```diff
@@ -62,12 +62,13 @@ const market = async () => {
         if (this.activeStall) {
           products = products.filter(p => p.stall_id == this.activeStall)
         }
-        if (!this.searchText || this.searchText.length < 2) return products
+        const searchText = this.searchText.toLowerCase()
+        if (!searchText || searchText.length < 2) return products
```
Comment (motorina0) [inline static/js/market.js:66]: if `searchText` can be `undefined`/`null` then move `const searchText = this.searchText.toLowerCase()` after the `if`

**Pull request merged**